docs/Behaviour - Client Side.md

```diff
@@ -13,14 +13,15 @@ Constraint Sets of Active Constraints SHOULD represent a consensus among Receive
 Constraint Sets of Receiver Capabilities with `urn:x-nmos:cap:meta:enabled` set to false MUST be ignored completely while making the processing.
 
 
-## Dynamic format changes on Sender
+## Dynamic Update of Resources
 
-### Changes which break Active Constraints
+The Controller MUST be capable of using Stream Compatibility API in conjunction with IS-04 APIs to discover and dynamically report the state of any relevant resources, including Inputs, Outputs, Senders and Receivers.
 
-NMOS Controller MUST track Sender's `subscription`'s `active` property during and after the IS-05 connection. If it became `false` unexpectedly, NMOS Controller SHOULD `GET /status` to figure out the reason of breaking the connection. If the State of the Sender is `active_constraints_violation`, then NMOS Controller SHOULD either disconnect all the Receivers or `PUT /constraints/active` once again and re-enable Sender via IS-05.
-
-### Changes which do not break Active Constraints
-
-Sender's changes which meet the Active Constraints do not affect the `active` property so if the transport file of the Sender has changed, the NMOS Controller SHOULD `PATCH /staged` of all the connected Receivers with the new `transport_file` requesting immediate activation.
+In order to avoid polling of the Stream Compatibility API, it is RECOMMENDED that Controllers use the WebSocket subscription mechanism of IS-04 Query API.
+* The Controller MUST indicate statuses of Inputs, Outputs, Senders and Receivers.
+* The Controller MUST report to the User when `status` attribute of the state of a Sender becomes `active_constraints_violation` and SHOULD report when it becomes `no_essence` or `awaiting_essence`.
+* The Controller SHOULD report to the User when `status` attribute of the state of an active Receiver becomes `non_compliant_stream`.
+* The Controller SHOULD `PATCH /staged` of all the connected Receivers with the new `transport_file` requesting immediate activation when the related active Sender updates the transport file.
+* The Controller MUST reflect changes in presence/absence of Inputs and Outputs to the User after a maximum of 30 seconds.
```

Comment:
```suggestion
* The Controller MUST indicate statuses of Inputs, Outputs, Senders and Receivers and relationship (or its absence) between Inputs/Outputs and Senders/Receivers listed in Stream Compatibility APIs.
* The Controller MUST indicate whether each Sender/Receiver discovered via IS-04 APIs is present in Stream Compatibility API
* The Controller MUST report to the User when `status` attribute of the state of a Sender becomes `active_constraints_violation` and SHOULD report when it becomes `no_essence` or `awaiting_essence`.
* The Controller SHOULD report to the User when `status` attribute of the state of an active Receiver becomes `non_compliant_stream`.
* The Controller SHOULD `PATCH /staged` of all the connected Receivers with the new `transport_file` requesting immediate activation when the related active Sender updates the transport file.
* The Controller MUST reflect changes in presence/absence of Inputs and Outputs to the User after a maximum of 30 seconds.
```

Reply:
awaiting_essence is a transient state and MUST not be reported to the user.

Reply:
"when `status` attribute of the state of a Sender" => seems to be "when `state` attribute of the status of a Sender"

Reply:
```suggestion
* The Controller MUST indicate statuses of Inputs, Outputs, Senders and Receivers and relationship (or its absence) between Inputs/Outputs and Senders/Receivers listed in Stream Compatibility APIs.
* The Controller MUST indicate whether each Sender/Receiver discovered via IS-04 APIs is present in Stream Compatibility API
* The Controller MUST report to the User when `state` attribute of the status of a Sender becomes `active_constraints_violation` and SHOULD report when it becomes `no_essence`.
* The Controller SHOULD report to the User when `state` attribute of the status of an active Receiver becomes `non_compliant_stream`.
* The Controller SHOULD `PATCH /staged` of all the connected Receivers with the new `transport_file` requesting immediate activation when the related active Sender updates the transport file.
* The Controller MUST reflect changes in presence/absence of Inputs and Outputs to the User after a maximum of 30 seconds.
```
Removed `awaiting_essence`. Should we dictate that Controller MUST not report about it?

Reply:
"The Controller MUST indicate statuses of Inputs, Outputs, Senders and Receivers and relationship (or its absence) between Inputs/Outputs and Senders/Receivers listed in Stream Compatibility APIs."

It seems to me that "relationship" is not precise enough. I would phrase it as follow instead:

"The Controller MUST indicate statuses of Inputs, Outputs, Senders and Receivers and SHOULD indicate the Inputs associated to a given Sender and Outputs associated to a given Receiver for Senders/Receivers listed in Stream Compatibility APIs."

The word "relationship" in general could have meant to show the interconnection of inputs with senders in general and outputs with receivers in general and I think that this is out of scope for IS-11.

Reply:
```suggestion
* The Controller MUST indicate statuses of Inputs, Outputs, Senders and Receivers and relationship (or its absence) between Inputs/Outputs and Senders/Receivers listed in Stream Compatibility APIs.
* The Controller MUST indicate whether each Sender/Receiver discovered via IS-04 APIs is present in Stream Compatibility API
* The Controller MUST report to the User when `state` attribute of the status of a Sender remains `active_constraints_violation` or `no_essence` after a maximum of 30 seconds.
* The Controller SHOULD report to the User when `state` attribute of the status of an active Receiver becomes `non_compliant_stream`.
* The Controller SHOULD `PATCH /staged` of all the connected Receivers with the new `transport_file` requesting immediate activation when the related active Sender updates the transport file.
* The Controller MUST reflect changes in presence/absence of Inputs and Outputs to the User after a maximum of 30 seconds.
```
Added the 30 seconds timeout for reporting about `state` of a Sender.

Reply:
>The Controller MUST indicate statuses of Inputs, Outputs, Senders and Receivers and SHOULD indicate the Inputs associated to a given Sender and Outputs associated to a given Receiver for Senders/Receivers listed in Stream Compatibility APIs.

It may be read as if orphan Inputs/Outputs can be ignored at all. Does the following sound more precise?
>The Controller MUST indicate statuses of Inputs, Outputs, Senders and Receivers listed in Stream Compatibility APIs and SHOULD indicate Inputs associated to a given Sender and Outputs associated to a given Receiver for such Senders/Receivers.